autoload/go/complete.vim

```diff
@@ -19,25 +19,25 @@ function! s:gocodeCommand(cmd, preargs, args) abort
     return
   endif
 
-  " we might hit cache problems, as gocode doesn't handle well different
-  " GOPATHS: https://github.com/nsf/gocode/issues/239
-  let old_gopath = $GOPATH
+  " We might hit cache problems, as gocode doesn't handle different GOPATHs
+  " well. See: https://github.com/nsf/gocode/issues/239
   let old_goroot = $GOROOT
-  let $GOPATH = go#path#Detect()
   let $GOROOT = go#util#env("goroot")
 
-  let socket_type = get(g:, 'go_gocode_socket_type', s:sock_type)
-  let cmd = printf('%s -sock %s %s %s %s',
-        \ go#util#Shellescape(bin_path),
-        \ socket_type,
-        \ join(a:preargs),
-        \ go#util#Shellescape(a:cmd),
-        \ join(a:args)
-        \ )
-
-  let result = go#util#System(cmd)
-  let $GOPATH = old_gopath
-  let $GOROOT = old_goroot
+  try
+    let socket_type = get(g:, 'go_gocode_socket_type', s:sock_type)
+    let cmd = printf('%s -sock %s %s %s %s',
+          \ go#util#Shellescape(bin_path),
+          \ socket_type,
+          \ join(a:preargs),
+          \ go#util#Shellescape(a:cmd),
+          \ join(a:args)
+          \ )
+
+    let result = go#util#System(cmd)
```

Comment:
This seems like a good time to convert this to `go#util#Exec`, but I won't object if you don't want to do it now.

Reply:
I'd rather not, since much of this isn't tested and can introduce regressions. Unless we're very careful we may have issues with double shell quoting for example.

I want to add a test for every tool which tests stuff like filenames with spaces, directories that don't exist, etc. and also fix whatever is needed (add build tags, use `#Exec()`, and whatever else might be needed). This is why I made the changes to the testing script and removed all the GOPATH detection.